crates/ty_python_semantic/resources/mdtest/narrow/assignment.md

```diff
@@ -75,7 +75,7 @@ class _:
 
     if cond():
         a = A()
-    reveal_type(a.x)  # revealed: int | None
+    reveal_type(a.x)  # revealed: int | None | Unknown
```

Comment:
This is the one part of this PR I'm really not sure about. Previously we weren't applying the global fallback (because `a` is possibly bound), but we were still walking enclosing scopes (because `scope.is_function_like(db)` used to be part of the `Unbound` short-circuit in `infer_place_load`). I _think_ that caused us to look up `a.x` here from an enclosing snapshot, and now we don't?

Reply:
Yes, that looks likely, and we apply the Unknown-widening to the global `A` but not the function-local one.